tasks/build.js

```diff
@@ -59,9 +59,12 @@ module.exports = function (grunt) {
     grunt.registerTask("full-build", ["git", "create-project", "build-www", "build", "stage", "package"]);
     grunt.registerTask("installer", ["set-release-optional", "full-build", "build-installer"]);
 
+    //Not to be used for MacOS
+    var target_arch_hint = grunt.option('arch') || undefined;
```

Comment:
grunt.option should return undefined in case no args are passed, so this is superfluous.